src/styles/index.scss

```diff
@@ -2,2 +1,75 @@
+body,
+html {
   margin: 0;
+  height: 100%;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+$calendar-size: 100px;
+$calendar-gap: $calendar-size / 100;
+$calendar-border: $calendar-size / 100;
+$calendar-padding: $calendar-size / 10;
+$calendar-font-size: 0.3 * $calendar-size;
+$calendar-animation-size: 0.2 * $calendar-size;
+$calendar-flex-width: (
+  (7 * $calendar-size) + ($calendar-gap * 6) + ($calendar-padding * 2)
+);
+$week-days: '-mon', '-tue', '-wed', '-thu', '-fri', '-sat', '-sun';
+$counter: 0;
+$base-spacing: $calendar-size + $calendar-gap;
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  padding: $calendar-padding;
+  width: $calendar-flex-width;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #eee;
+    width: $calendar-size;
+    height: $calendar-size;
+    border: $calendar-border solid black;
+    transition: 0.5s ease;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: $calendar-font-size;
+      }
+    }
+
+    &:hover {
```

Comment:
It's better to avoid using magic numbers. The color used for `background-color` in `&:hover` should be stored in a variable for better maintainability and readability of the code.